great_expectations/data_context/store/inline_store_backend.py

```diff
@@ -100,6 +100,10 @@ def _set(self, key: tuple[str, ...], value: Any, **kwargs: dict) -> None:
 
         if resource_type is DataContextVariableSchema.ALL_VARIABLES:
             config_commented_map_from_yaml = yaml.load(value)
+            # TODO: instead prevent these from being set??
+            # TODO: revert this before merge
+            for name in config_commented_map_from_yaml.get("fluent_datasources", {}):
+                config_commented_map_from_yaml["datasources"].pop(name)
```

Comment:
When we call `self._datasource_store.set(key=None, value=datasource)` this causes the `fluent` datasources to be added to the `datasources` config entry, which will then be validated using the old "Datasource" marshmallow schemas.

This temporary hack is just to demonstrate the problem.